h5netcdf/core.py

```diff
@@ -150,6 +150,15 @@ def __array__(self, *args, **kwargs):
         return self._h5ds.__array__(*args, **kwargs)
 
     def __getitem__(self, key):
+        if h5py.__version__ >= LooseVersion("3.0.0") and self._root.decode_strings:
+            if self.dtype == str:
+                return self._h5ds.asstr()[key]
+            if self.dtype == object:
+                if isinstance(self._h5ds[key], bytes) or (
+                    isinstance(self._h5ds[key], np.ndarray)
+                    and type(self._h5ds[key].flat[0]) == bytes
+                ):
+                    return self._h5ds.asstr()[key]
```

Comment:
I think this would work more reliable, by using h5py's API to check for string dtypes:
```suggestion
            if h5py.check_string_dtype(self._h5ds[key].dtype) is not None:
                return self._h5ds.asstr()[key]
```

Reply:
This took a while. We need to wrap `self._h5ds[key]` with `np.array()` to evaluate `.dtype` in any case.